packages/react/kibana_context/theme/theme_provider.tsx

```diff
@@ -50,14 +55,43 @@ export const KibanaThemeProvider = ({
   const kibanaTheme = useObservable(theme$, defaultTheme);
   const colorMode = useMemo(() => getColorMode(kibanaTheme), [kibanaTheme]);
 
-  // We have to add a breakpoint provider, because the `EuiProvider` we were using-- instead
-  // of `EuiThemeProvider`-- adds a breakpoint.  Without it here now, several Kibana layouts
-  // break, particularly sidebars.
-  //
-  // We can investigate removing it later, but I'm adding it here for now.
-  return (
-    <EuiThemeProvider {...{ colorMode, theme, ...props }}>
-      <CurrentEuiBreakpointProvider>{children}</CurrentEuiBreakpointProvider>
-    </EuiThemeProvider>
-  );
+  return <EuiThemeProvider {...{ colorMode, theme, ...props }}>{children}</EuiThemeProvider>;
 };
+
+/**
+ * Unfortunately, a lot of plugins are using `KibanaThemeProvider` without a parent
+ * `EuiProvider` which provides very necessary setup (e.g. Emotion cache, breakpoints).
+ *
+ * If so, we need to render an EuiProvider first (but without global styles, since those
+ * are already handled by `KibanaRootContextProvider`)
+ *
+ * TODO: We can hopefully remove this and revert to only exporting the above component once
+ * all plugins are on `KibanaRootContextProvider`?
+ */
+const KibanaThemeProviderCheck = ({ theme, children, ...props }: KibanaThemeProviderProps) => {
+  const hasEuiProvider = useIsNestedEuiProvider();
+
+  if (hasEuiProvider) {
+    return (
+      <KibanaThemeProviderOnly theme={theme} {...props}>
+        {children}
+      </KibanaThemeProviderOnly>
+    );
+  } else {
+    emitEuiProviderWarning(
+      'Your plugin used an KibanaThemeProvider without a required parent KibanaRootContextProvider.'
+    );
```

Comment:
@clintandrewhall - I'm not totally sure if you'd prefer an optional warning here or not or if this copy even makes sense.

I can revert https://github.com/elastic/kibana/pull/163103/commits/e3cd4f40be1323fa61208f05619989b947cc83ca if you'd prefer not :)